File: Week3/Code/get_TreeHeight.py
import csv
import math
import sys

#Calculate the distance
def TreeHeight(degrees, distance):
    """Find x to the power of 0.5."""
    radians = degrees*math.pi/180.0
    distance * math.tan(radians)
    height = distance * math.tan(radians)
    return (height)


#get command line arguments
inputfile = sys.argv[1]
outputfile = inputfile.replace('.csv', '')
outputfile = outputfile.replace('../Data/', '')
outputfile = "../Results/" + outputfile + "_treeheights_python.csv"

#Read in file
f = open(inputfile,'rb')
csvread = csv.reader(f)
header = next(csvread)


#prepare write file
g = open(outputfile,'wb')
csvwrite = csv.writer(g)

#calculate height and write output to file
csvwrite.writerow(header + ["Tree.Height.m"])
for row in csvread:
    tree_height = TreeHeight(float(row[2]),float(row[1]))
    csvwrite.writerow(row + [str(tree_height)])

f.close()
g.close()

File: Week3/Code/get_TreeHeight.R
#This function calculates heights of trees from the angle of
#elevation and the distance from the base using the trigonometric
#formula height = distance * tan(radians)

#ARGUMENTS:
#degrees The angle of elevation
#distance The distance from base

#OUTPUT:
#The height of the tree, same units as "distance"

TreeHeight <- function(degrees, distance){
  radians <- degrees * pi / 180
  height <- distance * tan(radians)
  #print(paste("Tree height is:", height))
  return (height)
}

#Loads csv file from command line argument
infile <- commandArgs(trailingOnly = TRUE)[1]
MyData <- read.csv(infile, header = TRUE) # import with headers

#Calculate heights
Tree.Height.m <- TreeHeight(MyData$Angle.degrees, MyData$Distance.m)
Output <- cbind(MyData,Tree.Height.m)


#Writes csv file with label from command line argument
outfile <- gsub(".csv","",infile)
outfile <- gsub("../Data/","",outfile)
outfile <- paste("../Results/",outfile,"_treeheights.csv",sep="")
write.csv(Output, outfile, row.names=FALSE,quote = FALSE)
